postgres/server/init.sls

```diff
@@ -86,23 +86,43 @@ postgresql-config-dir:
     - require:
       - cmd: postgresql-cluster-prepared
 
-{%- if postgres.postgresconf %}
+{%- set db_port = salt['config.option']('postgres.port') %}
```

Comment:
Instead of the suggested:

```sls
{%- if salt['pillar.get']('postgres.port') %}
```

* `config.option` gets the value from the minion configuration first, otherwise the pillar (which then works correctly with the `postgres.manage` states)
* Same as the method used in the code for `salt.modules.postgres`